### PR TITLE
11538_poly_solve_symbols

### PR DESCRIPTION
**Pull request by ctefer**
Updated constructor to treat NumberSymbols as symbols. polys containing non-algebraic number symbols are now treated as ZZ. This allows solvers to solve polynomials containing these specialized constants. 

Tests created were checked against WA. A test that ensures an empty set is not produced when given the original issue parameters was also produced. The non-empty set is hard to check against WA and contains tuples that need to be solved. 

A test was updated for test_solveset because the solver method now simplifies the output even further than was originally asserted.

### CODE REVIEW

File: sympy/solvers/tests/test_solvers.py [hunk 1]
```diff
@@ -1792,3 +1793,17 @@ def test_issue_2840_8155():
...
+    assert solve(x**3 + 2*E) == [
+        -cbrt(2 * E),
+        cbrt(2)*cbrt(E)/2 - cbrt(2)*sqrt(3)*I*cbrt(E)/2,
+        cbrt(2)*cbrt(E)/2 + cbrt(2)*sqrt(3)*I*cbrt(E)/2]
+    assert solve([x + 4,y + E],x,y) == {x:-4,y:-E}
+    assert solve([x**2 + 4,y + E],x,y) == [(-2*I, -E), (2*I, -E)]
+
+    e1 = x - y**3 + 4
+    e2 = x + y + 4 + 4 * E
+    assert solve([e1,e2],x,y) != []
```
Comment (Shekharrajak) [inline sympy/solvers/tests/test_solvers.py]: What is the output for `solve([e1,e2],x,y)`?

Comment (jksuom) [inline sympy/solvers/tests/test_solvers.py]:
```
[((-4*E*(54*E + 3*sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) - 4*(54*E + 3*sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) - 2*sqrt(3 + 324*exp(2)) - 36*E - (54*E + 3*sqrt(3)*sqrt(1 + 108*exp(2)))**(1/3) - 3**(5/6)*I*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(1/3) + 12*3**(1/6)*I*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) + 12*3**(1/6)*E*I*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3))/((3**(2/3) - 3*3**(1/6)*I)*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3)), 1/((-1/2 - sqrt(3)*I/2)*(54*E + sqrt(108 + 11664*exp(2))/2)**(1/3)) - (-1/2 - sqrt(3)*I/2)*(54*E + sqrt(108 + 11664*exp(2))/2)**(1/3)/3), ((-4*E*(54*E + 3*sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) - 4*(54*E + 3*sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) - 2*sqrt(3 + 324*exp(2)) - 36*E - (54*E + 3*sqrt(3)*sqrt(1 + 108*exp(2)))**(1/3) - 12*3**(1/6)*E*I*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) - 12*3**(1/6)*I*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) + 3**(5/6)*I*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(1/3))/((3**(2/3) + 3*3**(1/6)*I)*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3)), -(-1/2 + sqrt(3)*I/2)*(54*E + sqrt(108 + 11664*exp(2))/2)**(1/3)/3 + 1/((-1/2 + sqrt(3)*I/2)*(54*E + sqrt(108 + 11664*exp(2))/2)**(1/3))), (-4*E - 4 - 3**(2/3)/(3*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(1/3)) + 6*3**(1/3)*E/(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3) + 3**(5/6)*sqrt(1 + 108*exp(2))/(3*(18*E + sqrt(3)*sqrt(1 + 108*exp(2)))**(2/3)), -(54*E + sqrt(108 + 11664*exp(2))/2)**(1/3)/3 + (54*E + sqrt(108 + 11664*exp(2))/2)**(-1/3))]
```

Comment (Shekharrajak) [inline sympy/solvers/tests/test_solvers.py]: We may get soln in simple form if we replace `NumberSymbol` with it's value : 

```
In [37]: e1
Out[37]: 
     3    
x - y  + 4

In [38]: e2
Out[38]: x + y + 4 + 4⋅ℯ

In [39]: e2_new = e2.evalf()

In [40]: e2_new
Out[40]: x + y + 14.8731273138362

In [41]: solve([e1, e2], x, y)
Out[41]: []

In [42]: solve([e1, e2_new], x, y)
Out[42]: 
[(-12.8079458780453, -2.06518143579091), (-15.9057180317317 - 2.04908046988193⋅ⅈ, 1.03259071789545 + 2.04908046988193⋅ⅈ), (-15.9057180317317 + 2.049
08046988193⋅ⅈ, 1.03259071789545 - 2.04908046988193⋅ⅈ)]

In [43]: nonlinsolve([e1, e2_new], x, y)
Out[43]: 
{(-12.8079458780453, -2.06518143579091), (-15.9057180317316 - 2.04908046988193⋅ⅈ, 1.03259071789545 + 2.04908046988193⋅ⅈ), (-15.9057180317316 + 2.049
08046988193⋅ⅈ, 1.03259071789545 - 2.04908046988193⋅ⅈ)}

```

Comment (ctefer) [inline sympy/solvers/tests/test_solvers.py]: jksuom/Shekharrajak - I wasn't able to verify that as being the correct answer. I think it is, but someone smarter with tuple solve should verify that for the test. I merely wanted to show that that test did not provide an empty set.

Comment (ctefer) [inline sympy/solvers/tests/test_solvers.py]: Shekharrajak - we had discussed having NumberSymbol provide the approximate answer through evalf(), but we had hoped to produce answers that were consistent with evaluation of symbols (say if we were to replace E with a symbol 'k')

**Comment by ctefer** [comment]
First pull request so I had to make some adjustments to my main "master" branch. I also didn't do my commit messages well. The pull request is still valid, though if I needed to update the pull request I'm not sure I would understand how yet.

File: sympy/solvers/tests/test_solvers.py (hunk 1 above)
Comment (jksuom) [inline sympy/solvers/tests/test_solvers.py]: I believe the result is correct, but, actually, the purpose of the test is to show that a result is obtained. Hence it should not be necessary to copy the long answer. I think that something like `assert len(solve([e1, e2], x, y)) == 3` (with a space after the commas) would suffice.

**Comment by jksuom** [comment]
You seem to have some difficulties with the PR. It now shows "from unknown repository" on top of this page. The wiki on [development workflow](https://github.com/sympy/sympy/wiki/Development-workflow#create-a-patch-file-or-pull-request-for-github) may contain some useful information though it is a bit outdated. Note, in particular, that the name of your branch (indicated by a red check mark) should appear somewhere.

**Comment by ctefer** [comment]
I have created a new pull request due to issues with the last request. This issue can be closed.

**Pull request closed without merging**